crates/ruff_python_ast/src/source_code/mod.rs

```diff
@@ -23,3 +27,214 @@ pub fn round_trip(code: &str, source_path: &str) -> Result<String, ParseError> {
     generator.unparse_suite(&python_ast);
     Ok(generator.generate())
 }
+
+/// Gives access to the source code of a file and allows mapping between [`Location`] and byte offsets.
+#[derive(Debug)]
+pub struct SourceCode<'src, 'index> {
+    text: &'src str,
+    index: &'index LineIndex,
+}
+
+impl<'src, 'index> SourceCode<'src, 'index> {
+    pub fn new(content: &'src str, index: &'index LineIndex) -> Self {
+        Self {
+            text: content,
+            index,
+        }
+    }
+
+    /// Take the source code up to the given [`Location`].
+    pub fn up_to(&self, location: Location) -> &'src str {
+        let offset = self.index.location_offset(location, self.text);
+        &self.text[TextRange::up_to(offset)]
+    }
+
+    /// Take the source code after the given [`Location`].
+    pub fn after(&self, location: Location) -> &'src str {
```

Comment:
A bold rename!

Reply:
😅 I don't feel too strongly about it. I just found that `after` works better with `TextRange::up_to`.

Reply:
Hah me neither. I had used `until` and `after` to mimic the iterator API but I genuinely don't feel strongly, trust you to do whatever makes sense.